### PR TITLE
Introducing new Cluster Group Affinity Rule commands

### PR DESCRIPTION
**Pull request by robinharwood**
# PR Summary

Adding new command to the module.

## PR Checklist

- [x] **Descriptive Title:** This PR's title is a synopsis of the changes it proposes.
- [x] **Summary:** This PR's summary describes the scope and intent of the change.
- [x] **Contributor's Guide:** I have read the [contributors guide][contrib].
- [x] **Style:** This PR adheres to the [style guide][style].

[contrib]: https://learn.microsoft.com/powershell/scripting/community/contributing/overview
[style]: https://learn.microsoft.com/powershell/scripting/community/contributing/powershell-style-guide

### CODE REVIEW

**Comment by get-itips** [comment]
@scanum seems is still a draft

File: docset/winserver2022-ps/failoverclusters/Add-ClusterGroupToAffinityRule.md [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+### InputObject (cdxml)
+
+```
+Add-ClusterGroupToAffinityRule -InputObject <CimInstance[]> [-Groups] <String[]>
+[-CimSession <CimSession[]>] [-ThrottleLimit <Int32>] [-AsJob] [-PassThru] [<CommonParameters>]
+```
+
+## DESCRIPTION
+
+Adds a cluster group to a named affinity rule.
```
Comment (michaeltlombardi) [inline docset/winserver2022-ps/failoverclusters/Add-ClusterGroupToAffinityRule.md:34]: Can we extend this description? Currently, it's the same as the synopsis.

File: docset/winserver2022-ps/failoverclusters/Add-ClusterGroupToAffinityRule.md [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+
+## EXAMPLES
+
+### Example 1
```
Comment (michaeltlombardi) [inline docset/winserver2022-ps/failoverclusters/Add-ClusterGroupToAffinityRule.md:38]: Missing descriptive name for the example.

File: docset/winserver2022-ps/failoverclusters/Add-ClusterGroupToAffinityRule.md [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+
+Runs the cmdlet as a background job. Use this parameter to run commands that take a long time to
+complete.
+
+The cmdlet immediately returns an object that represents the job and then displays the command
+prompt. You can continue to work in the session while the job completes. To manage the job, use the
+`*-Job` cmdlets. To get the job results, use the
+[Receive-Job](https://go.microsoft.com/fwlink/?LinkID=113372) cmdlet.
+
+For more information about Windows PowerShell background jobs, see
```
Comment (michaeltlombardi) [inline docset/winserver2022-ps/failoverclusters/Add-ClusterGroupToAffinityRule.md:58]:
```suggestion
For more information about PowerShell background jobs, see
```

File: docset/winserver2022-ps/failoverclusters/Add-ClusterGroupToAffinityRule.md [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+Required: True
+Position: 1
+Default value: None
+Accept pipeline input: False
+Accept wildcard characters: False
+```
+
+### -InputObject
+
+Specifies the input object that is used in a pipeline command.
```
Comment (michaeltlombardi) [inline docset/winserver2022-ps/failoverclusters/Add-ClusterGroupToAffinityRule.md:110]: Can we specify more information about the input object for the user?

File: docset/winserver2022-ps/failoverclusters/Add-ClusterGroupToAffinityRule.md [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+Required: True
+Position: Named
+Default value: None
+Accept pipeline input: True (ByValue)
+Accept wildcard characters: False
+```
+
+### -Name
+
+The Affinity rule in question that groups will be added to.
```
Comment (michaeltlombardi) [inline docset/winserver2022-ps/failoverclusters/Add-ClusterGroupToAffinityRule.md:126]:
```suggestion
The Affinity rule to add the groups to.
```

File: docset/winserver2022-ps/failoverclusters/Add-ClusterGroupToAffinityRule.md [hunk 6]
```diff
@@ -0,0 +1,200 @@
...
+Position: 0
+Default value: None
+Accept pipeline input: True (ByPropertyName)
+Accept wildcard characters: False
+```
+
+### -PassThru
+
+Returns an object representing the item with which you are working. By default, this cmdlet doesn't
+generate any output.
```
Comment (michaeltlombardi) [inline docset/winserver2022-ps/failoverclusters/Add-ClusterGroupToAffinityRule.md:143]: Can this be more descriptive of the output object and include its type for the user?

File: docset/winserver2022-ps/failoverclusters/Add-ClusterGroupToAffinityRule.md [hunk 7]
```diff
@@ -0,0 +1,200 @@
...
+
+### System.String[]
+
+### Microsoft.Management.Infrastructure.CimInstance[]
+
+## OUTPUTS
+
+### Microsoft.Management.Infrastructure.CimInstance
+
+### Microsoft.Management.Infrastructure.CimInstance#root/MSCLUSTER/MSCluster_AffinityRule
```
Comment (michaeltlombardi) [inline docset/winserver2022-ps/failoverclusters/Add-ClusterGroupToAffinityRule.md:194]: Can we add descriptions for these types, when and how they're used and returned?

File: docset/winserver2022-ps/failoverclusters/Remove-ClusterGroupFromAffinityRule.md [hunk 8]
```diff
@@ -0,0 +1,200 @@
...
+ [-CimSession <CimSession[]>] [-ThrottleLimit <Int32>] [-AsJob] [-PassThru] [<CommonParameters>]
+```
+
+## DESCRIPTION
+
+Removes a cluster group from an affinity rule. This doesn't remove the rule itself.
+
+## EXAMPLES
+
+### Example 1
```
Comment (michaeltlombardi) [inline docset/winserver2022-ps/failoverclusters/Remove-ClusterGroupFromAffinityRule.md:38]: Missing descriptive name after the example number.

File: docset/winserver2022-ps/failoverclusters/Remove-ClusterGroupFromAffinityRule.md [hunk 9]
```diff
@@ -0,0 +1,200 @@
...
+
+```powershell
+Remove-ClusterGroupFromAffinityRule -Name MuRyle -Groups MyGroup -Cluster MyCluster
+```
+
+This example removes the cluster group MyGroup from the affinity rule named MyRule.
```
Comment (michaeltlombardi) [inline docset/winserver2022-ps/failoverclusters/Remove-ClusterGroupFromAffinityRule.md:44]:
```suggestion
This example removes the cluster group `MyGroup` from the affinity rule named `MyRule`.
```

File: docset/winserver2022-ps/failoverclusters/Add-ClusterGroupToAffinityRule.md [hunk 10]
```diff
@@ -0,0 +1,200 @@
...
+
+## EXAMPLES
+
+### Example 1
+
+```powershell
+Add-ClusterGroupToAffinityRule -Groups MyGroup -Name MyRule -Cluster MyCluster
+```
+
+This example adds the group named MyGroup to the affinity rule named MyRule on the cluster named.
```
Comment (michaeltlombardi) [inline docset/winserver2022-ps/failoverclusters/Add-ClusterGroupToAffinityRule.md:44]:
```suggestion
This example adds the group named `MyGroup` to the affinity rule named `MyRule` on the cluster named `MyCluster`.
```

File: docset/winserver2022-ps/failoverclusters/Remove-ClusterGroupFromAffinityRule.md [hunk 11]
```diff
@@ -0,0 +1,200 @@
...
+
+Runs the cmdlet as a background job. Use this parameter to run commands that take a long time to
+complete.
+
+The cmdlet immediately returns an object that represents the job and then displays the command
+prompt. You can continue to work in the session while the job completes. To manage the job, use the
+`*-Job` cmdlets. To get the job results, use the
+[Receive-Job](https://go.microsoft.com/fwlink/?LinkID=113372) cmdlet.
+
+For more information about Windows PowerShell background jobs, see
```
Comment (michaeltlombardi) [inline docset/winserver2022-ps/failoverclusters/Remove-ClusterGroupFromAffinityRule.md:58]:
```suggestion
For more information about PowerShell background jobs, see
```

File: docset/winserver2022-ps/failoverclusters/Remove-ClusterGroupFromAffinityRule.md [hunk 12]
```diff
@@ -0,0 +1,200 @@
...
+Required: False
+Position: Named
+Default value: None
+Accept pipeline input: False
+Accept wildcard characters: False
+```
+
+### -Groups
+
+The groups to be removed from the affinity rule.
```
Comment (michaeltlombardi) [inline docset/winserver2022-ps/failoverclusters/Remove-ClusterGroupFromAffinityRule.md:94]:
```suggestion
The groups to remove from the affinity rule.
```

File: docset/winserver2022-ps/failoverclusters/Remove-ClusterGroupFromAffinityRule.md [hunk 13]
```diff
@@ -0,0 +1,200 @@
...
+Required: True
+Position: 1
+Default value: None
+Accept pipeline input: False
+Accept wildcard characters: False
+```
+
+### -InputObject
+
+Specifies the input object that is used in a pipeline command.
```
Comment (michaeltlombardi) [inline docset/winserver2022-ps/failoverclusters/Remove-ClusterGroupFromAffinityRule.md:110]: Can this be more specific to the cmdlet?

File: docset/winserver2022-ps/failoverclusters/Remove-ClusterGroupFromAffinityRule.md [hunk 14]
```diff
@@ -0,0 +1,200 @@
...
+Required: True
+Position: Named
+Default value: None
+Accept pipeline input: True (ByValue)
+Accept wildcard characters: False
+```
+
+### -Name
+
+The name of the affinity rule to remove groups from
```
Comment (michaeltlombardi) [inline docset/winserver2022-ps/failoverclusters/Remove-ClusterGroupFromAffinityRule.md:126]:
```suggestion
The name of the affinity rule to remove groups from.
```

File: docset/winserver2022-ps/failoverclusters/Remove-ClusterGroupFromAffinityRule.md [hunk 15]
```diff
@@ -0,0 +1,200 @@
...
+Required: False
+Position: 0
+Default value: None
+Accept pipeline input: True (ByPropertyName)
+Accept wildcard characters: False
+```
+
+### -PassThru
+
+Returns an object representing the item with which you are working. By default, this cmdlet doesn't
```
Comment (michaeltlombardi) [inline docset/winserver2022-ps/failoverclusters/Remove-ClusterGroupFromAffinityRule.md:142]: Can this be more descriptive of the output object and include its type?

File: docset/winserver2022-ps/failoverclusters/Remove-ClusterGroupFromAffinityRule.md [hunk 16]
```diff
@@ -0,0 +1,200 @@
...
+Position: Named
+Default value: None
+Accept pipeline input: False
+Accept wildcard characters: False
+```
+
+### -ThrottleLimit
+
+Specifies the maximum number of concurrent operations that can be established to run the cmdlet. If
+this parameter is omitted or a value of `0` is entered, then Windows PowerShell&reg; calculates an
```
Comment (michaeltlombardi) [inline docset/winserver2022-ps/failoverclusters/Remove-ClusterGroupFromAffinityRule.md:160]:
```suggestion
this parameter is omitted or a value of `0` is entered, then PowerShell calculates an
```

File: docset/winserver2022-ps/failoverclusters/Remove-ClusterGroupFromAffinityRule.md [hunk 17]
```diff
@@ -0,0 +1,200 @@
...
+
+### System.String[]
+
+### Microsoft.Management.Infrastructure.CimInstance[]
+
+## OUTPUTS
+
+### Microsoft.Management.Infrastructure.CimInstance
+
+### Microsoft.Management.Infrastructure.CimInstance#root/MSCLUSTER/MSCluster_AffinityRule
```
Comment (michaeltlombardi) [inline docset/winserver2022-ps/failoverclusters/Remove-ClusterGroupFromAffinityRule.md:194]: Can we add descriptions for these types - when and how they're used or returned?

File: docset/winserver2022-ps/failoverclusters/Add-ClusterGroupToAffinityRule.md [hunk 18]
```diff
@@ -0,0 +1,200 @@
...
+Position: Named
+Default value: None
+Accept pipeline input: False
+Accept wildcard characters: False
+```
+
+### -ThrottleLimit
+
+Specifies the maximum number of concurrent operations that can be established to run the cmdlet. If
+this parameter is omitted or a value of `0` is entered, then Windows PowerShell&reg; calculates an
```
Comment (michaeltlombardi) [inline docset/winserver2022-ps/failoverclusters/Add-ClusterGroupToAffinityRule.md:160]:
```suggestion
this parameter is omitted or a value of `0` is entered, then PowerShell calculates an
```

**Review by michaeltlombardi — CHANGES_REQUESTED** [review]
Minor corrections and suggestions for further documentation.

**Review by michaeltlombardi — APPROVED** [review]

**Comment by michaeltlombardi** [comment]
@robinharwood - happy to merge this one if you're finished with it.

**Comment by robinharwood** [comment]
Yes please @michaeltlombardi, I was waiting for the first PR so that link wasn't broken 😄

**Pull request merged**